File: 算法/代码随想录算法PDF/00调试源代码/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        option = {}
        for i in range(len(nums)):
            need = target - nums[i]
            if need in option:
                return [option[need], i]
            else:
                option[nums[i]] = i
        return []
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> option;
        for (int i = 0; i < nums.size(); i++) {
            int need = target - nums[i];  // 要找的值
            if (option.count(need)) {
                return {option[need], i};
            } else {
                option[nums[i]] = i;
            }
        }
        return {};
    }
};
// @lc code=end

